### PR TITLE
feat: implemented pokemon overview page

### PR DESCRIPTION
**Pull request by jayveeannsoriano**
- deleted home_page.dart
- created vm and connector for pokemon overview page state and UI
- displayed the pokemon list as a 2 grid view in pokemon overview page
- modified main.dart to display the newly added page
- created reusable widgets and string extension
- modified widget_test.dart to resolve flutter lint issues

- pokemon overview page UI
image

- flutter lint analyzer results and format
image

### CODE REVIEW

File: lib/features/pokemon_overview/pokemon_overview_page.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  });
+
+  final List<Pokemon> pokemons;
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: pokemonOverviewTitle,
```
Comment (shawn-ffufm) [inline lib/features/pokemon_overview/pokemon_overview_page.dart]: no need for extra trailing comma 🙂

File: lib/features/pokemon_overview/pokemon_overview_page.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        ),
+        body: GridView.builder(
+          padding: gridViewPadding,
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: gridCount),
+          itemCount: pokemons.length,
+          itemBuilder: (context, index) {
```
Comment (shawn-ffufm) [inline lib/features/pokemon_overview/pokemon_overview_page.dart]: rename `context` to `_` since it's unused 🙂

File: lib/utils/constants.dart [hunk 3]
```diff
@@ -1,5 +1,19 @@
+import 'package:flutter/material.dart';
+
 const String pokemonOffSet = '0';
 const String pokemonLimit = '151';
 
 //TODO: added for testing GetPokemonDetailsAction. This will be removed
 const pokemonNameInput = 'jigglypuff';
+
+const pokemonOverviewTitle = Text('Pokemon Overview');
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart]: make it only a String not a Text widget 😅

File: lib/utils/constants.dart [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+const gridViewPadding = EdgeInsets.all(10.0);
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart]: seems no need for this 🙂

File: lib/utils/constants.dart [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
+const int gridCount = 2;
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart]: same here 🙂

File: lib/utils/constants.dart [hunk 6]
```diff
@@ -1,5 +1,19 @@
...
+const urlSplitter = '/';
+const int idIndex = 6;
+const indexUrlReplace = 'index';
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart]: kindly rename this to `indexKey` 🙂

File: lib/utils/constants.dart [hunk 7]
```diff
@@ -1,5 +1,19 @@
...
+const double imageWidth = 200.0;
+const double imageHeight = 130.0;
+const double gridContentSpacing = 20.0;
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart:15]: seems no need for this 🙂

File: lib/utils/constants.dart [hunk 8]
```diff
@@ -1,5 +1,19 @@
...
+const pokemonImageUrlPath =
```
Comment (shawn-ffufm) [inline lib/utils/constants.dart:16]: don't forget `[index]` since that'll be the key to be replaced 😉

File: lib/widgets/pokemon_tile_card.dart [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import 'package:pokedex_async_redux/api/model/model.dart';
+import 'package:pokedex_async_redux/utils/constants.dart';
+import 'package:pokedex_async_redux/widgets/spacing.dart';
+import 'package:dartx/dartx.dart';
+import 'package:flutter/material.dart';
+
+class PokemonTileCard extends StatelessWidget {
+  const PokemonTileCard({
+    Key? key,
+    required this.pokemon,
```
Comment (shawn-ffufm) [inline lib/widgets/pokemon_tile_card.dart:9]: make it as a habit to place required properties on top of everything 🙂

File: lib/widgets/pokemon_tile_card.dart [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  }) : super(key: key);
+
+  final Pokemon pokemon;
+
+  @override
+  Widget build(BuildContext context) {
+    final pokemonId = pokemon.url.split(urlSplitter)[idIndex];
+    final pokemonImageUrl = pokemonImageUrlPath.replaceAll(indexUrlReplace, pokemonId);
+    final pokemonName = pokemon.name.capitalize();
```
Comment (shawn-ffufm) [inline lib/widgets/pokemon_tile_card.dart:19]: liked what you did above, it's neat 💪

File: lib/widgets/spacing.dart [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  const VerticalSpace({
+    Key? key,
+    required this.spacing,
+  }) : super(key: key);
+
+  final double spacing;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(height: spacing);
```
Comment (shawn-ffufm) [inline lib/widgets/spacing.dart]: just use fat arrow notation 🙂

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see pr feedback 🙂

File: lib/utils/constants.dart (hunk 7 above)
Comment (jayveeannsoriano) [reply lib/utils/constants.dart:15]: these values were used in pokemon_tile_card.dart 
image

File: lib/utils/constants.dart (hunk 5 above)
Comment (jayveeannsoriano) [reply lib/utils/constants.dart]: these values were used in pokemon_overview_page.dart 
image

File: lib/utils/constants.dart (hunk 4 above)
Comment (jayveeannsoriano) [reply lib/utils/constants.dart]: these values were used in pokemon_overview_page.dart
image

Comment (shawn-ffufm) [reply lib/utils/constants.dart]: yep, I meant no need to declare a constant just immediately place it there 🙂

File: lib/utils/constants.dart (hunk 5 above)
Comment (shawn-ffufm) [reply lib/utils/constants.dart]: same answer above 🙂

**Review by shawn-ffufm — APPROVED** [review]

**Pull request merged**